tutorials/i18n/localization_using_gettext.rst

```diff
@@ -1,179 +1,311 @@
-pybabel needs to process (note that we process GDScript as Python, which is
-generally sufficient):
+    var abilities = [tr("dodge"), tr("strike"), tr("shoot")]
 
-.. code-block:: none
+    func _ready():
+        print(tr("Hello Sir Godot"))
+        
+        for ability in abilities:
+            print(ability)
+        
+        print(tr_n("Found %d diamond", "Found %d diamonds", 10) % 10)
+        pass 
 
-    [python: **.gd]
-    encoding = utf-8
+2) Extract marked strings (Generate POT)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-    [godot_scene: **.tscn]
-    encoding = utf-8
+After we have marked the texts to be translated in ``GDScript`` and created some nodes containing texts, 
+we can now extract these strings. In the Editor, go to **Project**, **Project Settings...**, **Localization**, 
+**POT generation**, include the files containing the texts that need to be translated.
 
-You can then run pybabel like so:
+.. image:: img/pot_gen.png
 
-.. code-block:: shell
+After you have included the files, click **Generate POT** and select a location. This will generate a translation 
+template (POT) file in the location you have selected. POT file contains all the source strings that 
+need to be translated. You can open it with a text editor or a gettext editor, such as `Poedit <https://poedit.net/>`_.
 
-    pybabel extract -F babelrc -k text -k LineEdit/placeholder_text -k tr -o godot-l10n.pot .
+If you open it with a text editor, you'll see that the ``msgid`` field contains the strings that we want to translate. In the 
+next line after ``msgid``, we have ``msgstr`` field. ``msgstr`` will later contain the translated strings. Right now they are 
+empty because we haven't provided any translation yet.
 
-Use the ``-k`` option to specify what needs to be extracted. In this case,
-arguments to :ref:`tr() <class_Object_method_tr>` will be translated, as well
-as properties named "text" (commonly used by Control nodes) and LineEdit's
-"placeholder_text" property.
+3) Translate the strings (Translate PO)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Creating a messages file from a PO template
--------------------------------------------
+POT file is just a translation template. Before we can begin translating, we need to convert the POT file into a ``.po`` file, 
+which contains the translation for a specific language. For example, we can have ``fr.po`` (French), ``ar.po`` (Arabic), 
+``ru.po`` (Russian) etc. 
 
-The ``msginit`` command is used to turn a PO template into a messages file.
-For instance, to create a French localization file, use the following command
-while in the ``locale`` directory:
+To convert a POT file to a PO file, we have two options:
+
+- Open the POT file with a gettext editor, and the editor will ask us to create a new PO file with the language that we want.
+
+- Download gettext tools (the end of this page contains guide on installing gettext tools), and execute the command below in the 
```

Comment:
We can use a linked reference to make this easier to follow:

```suggestion
- :ref:`Download gettext tools <doc_localization_using_gettext_installing_tools>`, and execute the command below in the 
```

Make sure to apply the suggestion that adds the reference as well (it's below).